### DISCUSSION TITLE
using components in DataTable cells

### DISCUSSION THREAD

**Discussion by gabrielleecredera** [carbon-design-system/carbon-components-svelte#1686] [Q&A] [answered]
My understanding is that svelte doesn't support passing components around as props.

This means that we cannot add components into cells of the `<DataTable>` as the only way to specify rows is by passing a list of objects into the `rows` prop.
(i.e. you can't add links / images in a cell in DataTable, only text / numbers / booleans)

For that I have 2 questions:
1. is my understanding correct? there's currently no way to achieve that?
2. would we consider exporting the internal `<TableRow>` and `<TableCell>` (and other) components so that we can make use of slots to render components in cells? (this would take quite some work tho as the props of those components are currently fully controlled by DataTable.svelte which would need to be extracted from to function as an individual component)

**Comment by theetrain** [comment, marked as answer]
Hello, there is currently a way to add anything within DataTable cells via the named slot `cell`.

Docs: https://carbon-components-svelte.onrender.com/components/DataTable#slotted-cells
Example: https://svelte.dev/repl/d5ad7e17e80c4637856e5026a656c0a6?version=3.56.0

A current limitation is the presence of slotted content will not remove the cell's padding, but with css `:has` it should be achievable. Note that `:has` [isn't yet supported in Firefox](https://caniuse.com/css-has).

**Reply by gabrielleecredera** [reply]
ah somehow I missed that example.
very cool, thanks! :)
